awsshell/wizard.py

```diff
@@ -59,23 +111,23 @@ def __handle_static_retrieval(self):
         return self.retrieval.get('Resource')
 
     def __handle_request_retrieval(self):
-        # TODO very basic requests... refactor needed
         req = self.retrieval['Resource']
-        # initialize botocore session and client for service in the request
-        session = botocore.session.get_session()
-        client = session.create_client(req['Service'])
+        # get client from wizard's cache
+        client = self._wizard._cached_creator.create_client(req['Service'])
         # get the operation from the client
         operation = getattr(client, xform_name(req['Operation']))
         # get any parameters
         parameters = req.get('Parameters', {})
-        env_parameters = self._resolve_parameters(req.get('EnvParameters', {}))
+        env_parameters = \
+            self._wizard.env.resolve_parameters(req.get('EnvParameters', {}))
         # union of parameters and env_parameters, conflicts favor env_params
         parameters = dict(parameters, **env_parameters)
         # execute operation passing all parameters
         return operation(**parameters)
 
     def _handle_retrieval(self):
-        print(self.prompt)
+        # TODO decouple wizard from all I/O
+        sys.stdout.write(self.prompt+'\n')
```

Comment:
Sorry for being pedantic, but you'll need to flush as well. The reason for that is that some shells (e.g. cygwin) look like pipes to python, so all output will be buffered until it hits some size rather than being immediately flushed. The end result is that text won't be displayed when you think it will.